### ISSUE TITLE
Calls related to object creation in HaaS database fails 

### ISSUE THREAD

**Issue by SahilTikale** [CCI-MOC/hil#580]
Following is the behavior of the bug:

When trying to register new nodes or Users, it fails with the following error despite of giving correct inputs. 

INPUT: `haas node_register dummyNode01 ipmi host01 user01 pass1234`

OUTPUT:

```
Unexpected status code: 400
Response text:
{"msg": "The request body {u'obm': {u'host': u'host01', u'password': u'pass1234', u'type': 
u'http://schema.massopencloud.org/haas/v0/obm/ipmi', u'user': u'user01'}} is not valid for this request.", 
"type": "ValidationError"}
```

Other calls like: `haas user_create <username> <is_admin>` are also failing with the same error.

**Temporary Fix:**
For every api calls that are failing with the above described behavior do the following:

-- Modify the respective call in api.py and mention the parameter passed as URL explicitly in the json file.
**Example**
Modify the `node_register` call:
Original:

``` python
@rest_call('PUT', '/node/<node>', schema=Schema({
    'obm':{
        'type': basestring,
        Optional(object):object,
        },
}))
def node_register(node, **kwargs):
    """Create node.

    If the node already exists, a DuplicateError will be raised.
    The node is initially registered with no nics; see the method node_register_nic.
    """
```

Include the line `'node': basestring,` as follows:

``` python
@rest_call('PUT', '/node/<node>', schema=Schema({
    'node': basestring,
    'obm':{
        'type': basestring,
        Optional(object):object,
        },
}))
def node_register(node, **kwargs):
    """Create node.

    If the node already exists, a DuplicateError will be raised.
    The node is initially registered with no nics; see the method node_register_nic.
    """
```

After this change the code will work as expected.

**Comment by henn** [comment]
@ravisantoshgudimetla and I just finished running git bisect with `haas node_register node1 ...`, and it pointed to ed42146817e0ad4bc3a04e4272a8638f70b2c141

It looks like @zenhack hit something similar in 9400f6953aa03d8952f4de812245a3e184262e90

**Comment by henn** [comment]
FYI - I think that the automated apache tests that @knikolla is working on should catch these types of issues in the future.

**Comment by naved001** [comment]
Looks like this was fixed by #581 see commit [message here](https://github.com/CCI-MOC/hil/pull/581/commits/973a176c22b957b7e89a3280d6d3dd8a62f3f557).
